### DISCUSSION TITLE
2022-04-05 Developer Console Update

### DISCUSSION THREAD

**Discussion by DiscRisknBisque** [near/community#241] [Pagoda Platform]
# Summary

- On the product size, internal process creation and streamlining lays the foundation for Pagoda's continued growth.
- We've been reaching out to DAOs for feedback on their experiences managing and developing with their DAOs, if you'd like to participate, reach out to Josh Quintal below. Projects outside the NEAR ecosystem welcome!
- We're adding a feedback/support widget to Developer Console to allow you to provide feedback and report bugs more easily.
- Polish, bug fixes, and AstroDAO integration work continues as we prepare for a highly productive Q3.

> ℹ️ For a quick primer on Developer Console: check out Josh's ETH Denver presentation going over it at a high level, then specifically highlighting the interactive NFT Marketplace tutorial!
_
*Developer Console - Learn, Create, and Operate All in One Place*
[Presentation (16 Minutes)](https://www.youtube.com/watch?v=J-uFFEELi0s) | [Slides](https://docs.google.com/presentation/d/1sX4yrqNUxYSVEl3V5NOoyZUK0I-Sw0RDy9fRjJbNKBA/edit?usp=sharing)  
_  
[**Try Developer Console for yourself here.**](https://console.pagoda.co//)

### Product Feedback

You can reach Josh Quintal with product feedback, Dev Console or otherwise, here:

| Method | Josh (Dev Console & Explorer) |
| ------------- | ------------- |
| Calendly | [30 min Timeslot](https://calendly.com/josh-quintal/developer-console-user-interview) |
| Email | josh.quintal@near.org |
| GitHub | @OnlyOneJMJQ |
| Twitter | [@OnlyOneJMJQ](https://twitter.com/OnlyOneJMJQ) |
| Telegram | @OnlyOneJMJQ |
| Discord | OnlyOneJMJQ#4237 |

# Objectives & Key Results

*New quarter, new OKRs!* We'll be finalizing our next set of OKRs at our offsite next week. We set some overly ambitious ones last time, which will be rescoped and rollover to next quarter. Namely, the usage metrics.

Developer Console is a go-to tool for majority of NEAR developers
- [ ] `13%` Developer Console is used by 50% of developers
- [ ] `10%` 50% of RPC calls are using an API key
- [x] `(100%)` Deploy an interactive NFT tutorial to Dev Console based on the docs-NFT tutorial
- [ ] `10%` Build product plan for security scans/audits through Dev Console
- [ ] `90%` Data platform infrastructure supports future use and growth
- [ ] `35%` Deploy developer feedback mechanism
- [ ] `80%` Developer Console has all environments setup and CI/CD working
- [x] `(100%)` Developer Console expresses the new brand through identity and design

# Done

### Product & Marketing

Current in process product tasks are long-running, but have status updates:
- @OnlyOneJMJQ: Defining MVP for AstroDAO integration
  - Split DAO user interview sessions into two phases: exploratory and 
  - Review Phase 1 mockups
  - Prepare contract interaction examples for wireframing
- @OnlyOneJMJQ: Gather feedback on initial release of Developer Console
  - Interview 1/2 respondents
- @OnlyOneJMJQ: Internal sync on product vision

### Development

- @jon-lewis: Show mainnet key until we know it's a tutorial ([Framework#125](https://github.com/near/developer-console-framework/pull/125))
- @jon-lewis: Fix GitHub snippets ([Framework#124](https://github.com/near/developer-console-framework/pull/124))
- @jon-lewis: Fix snapshot tests ([Framework#122](https://github.com/near/developer-console-framework/pull/122))
- @jon-lewis: Run e2e and snapshot tests locally ([Framework#119](https://github.com/near/developer-console-framework/pull/119))
- @calebjacob: ES Version 2019 (Node 14) ([Framework#117](https://github.com/near/developer-console-framework/pull/117))
- @calebjacob: TS Lint Improvements ([Framework#116](https://github.com/near/developer-console-framework/pull/116))
- @calebjacob: Improved Types (V2) ([Framework#110](https://github.com/near/developer-console-framework/pull/110))
- @mpeterdev: Dev Console roadmapping with @OnlyOneJMJQ

# Doing

### Product & Marketing

- @OnlyOneJMJQ: Defining MVP for AstroDAO integration
  - Interviewing various DAOs about their experiences and needs
  - Preparing contract interaction examples for wireframing
- @OnlyOneJMJQ: Gather feedback on initial release of Developer Console
  - Interview 2/2 respondents
- @OnlyOneJMJQ: Long-term vision planning for Developer Console
  - @OnlyOneJMJQ: Catalog discussion topics for offsite
- @OnlyOneJMJQ: Organizing workshop for offsite in April (Ref?)

### Development

- @jon-lewis: Adding more snapshot tests
- @jon-lewis: Exploring potential font load optimizations
- @jon-lewis: Mutate SWR cache on project delete ([Framework#81](https://github.com/near/developer-console-framework/issues/81))
- @jon-lewis: Extract analytics calls to library to prepare for switch to segment ([Framework#77](https://github.com/near/developer-console-framework/issues/77))
- @jon-lewis: Gracefully handle users signing in with a different social provider than they registered with ([Framework#21](https://github.com/near/developer-console-framework/issues/21))
- @jon-lewis: Add unit testing ([Framework#12](https://github.com/near/developer-console-framework/issues/12))
- @jon-lewis: Working on implementation plan for Postgres audit tables
- @mpeterdev: Draft organization and team schema updates to prepare for AstroDAO
- @mpeterdev: Coordinate with Infrastructure team to coordinate updates to RPC service, including JWTs for frontend use
- @mpeterdev: Catalog discussions topics and whiteboard session candidates for offsite
- @mpeterdev: Research technical requirements for DAO integration
- @mpeterdev: Rework environment variable files ([Framework#11](https://github.com/near/developer-console-framework/issues/11))
- @mpeterdev: Review error states for SWR fetches ([Framework#6](https://github.com/near/developer-console-framework/issues/6))

**Comment by maxzaver** [comment]
> Polish, bug fixes, and AstroDAO integration work continues as we prepare for a highly productive Q3.

Did we mean to say Q2 here?
